src/Attribute/RouteParamValue/AbstractValues.php

```diff
@@ -6,10 +6,15 @@
 
 abstract class AbstractValues implements ValuesInterface
 {
+    /**
+     * @return non-empty-list<?scalar>
```

Comment:
I think we need to update the docs on `ValuesInterface::toArray()` as well.

Reply:
@HypeMC i'd leave it as is, because of `CompoundValues::toArray` which have nested structure of other values.